pyxem/generators/red_intensity_generator.py

```diff
@@ -0,0 +1,213 @@
+                                                  C_values, s_size, s_scale, type)
+        # self.fit = np.array(background.sum_squares).reshape(
+        #            self.nav_size[0],self.nav_size[1],self.sig_size[0])
+
+        self.normalisation = normalisation  # change this
+        self.background_fit = fit
+        return
+
+    def subtract_bkgd_pattern(self, bkgd_pattern):
+        """Subtracts a background pattern from the signal. This method will edit
+        self.signal.
+
+        Parameters
+        ----------
+        Bkgd_pattern : np.array
+                    A numpy array of a single line profile of the same resolution
+                    (same number of pixels) as the radial profile.
+        """
+        self.signal = self.signal - bkgd_pattern
+
+        return
+
+    def mask_from_bkgd_pattern(self, mask_pattern, mask_threshold=1):
+        """Uses a background pattern with a threshold, and sets that part of
+        the signal to zero, effectively adding a mask. This can be used to mask
+        the central beam. This method will edit self.signal.
+
+        Parameters
+        ----------
+        mask_pattern : np.array
+                    A numpy array line profile of the same resolution
+                    as the radial profile.
+        mask_threshold : int or float
+                    An integer or float threshold. Any pixel in the
+                    mask_pattern with lower intensity is kept, any with
+                    higher or equal is set to zero.
+        """
+
+        mask_array = mask_pattern < mask_threshold
+
+        self.signal = self.signal * mask_array.astype(float)
+
+        return
+
+    def get_reduced_intensity(self, s_cutoff=None):
+        """Obtains a reduced intensity profile from the radial profile.
+
+        Parameters
+        ----------
+        s_cutoff : list of float
+                    A list of the form [s_min, s_max] to change the s_cutoff
+                    from the fit.
+        """
+        if s_cutoff:
+            self.cutoff = s_cutoff
+        else:
+            s_cutoff = self.cutoff
+
+        # define numerical cutoff to remove certain data parts
+        s_scale = self.signal.axes_manager.signal_axes[0].scale
+        num_min, num_max = int(s_cutoff[0] / s_scale), int(s_cutoff[1] / s_scale)
+
+        s = np.arange(self.signal.axes_manager.signal_axes[0].size,
+                      dtype='float64')
+        s *= self.signal.axes_manager.signal_axes[0].scale
+        # remember axes scale and size!
+        reduced_intensity = (4 * np.pi * s *
+                             np.divide((self.signal.data - self.background_fit),
+                                       self.normalisation))
+
+        #ri = ReducedIntensityProfile(reduced_intensity.data[:,:,num_min:num_max])
+        ri = ReducedIntensityProfile(reduced_intensity)
+        ax_old = self.signal.axes_manager.navigation_axes
+        ri.axes_manager.navigation_axes[0].scale = ax_old[0].scale
```

Comment:
@pc494 didn't you add a function somewhere to tidy this up?

Reply:
https://github.com/pyxem/pyxem/blob/master/pyxem/signals/__init__.py has `transfer_navigation_axes()` and `transfer_signal_axes()`